bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java

```diff
@@ -370,11 +370,18 @@ public void modifyTable(TableName arg0, TableDescriptor arg1) throws IOException
   }
 
   @Override
-  public Future<Void> truncateTableAsync(TableName arg0, boolean arg1) throws IOException {
-    // TODO - implementable with async hbase2
-    throw new UnsupportedOperationException("truncateTableAsync");
+  public Future<Void> truncateTableAsync(TableName tableName, boolean preserveSplits) throws IOException {
+   if (!preserveSplits) {
+      LOG.info("truncate will preserveSplits. The passed in variable is ignored.");
+   }
+    return issueBulkDeleteAsync(tableName, DropRowRangeRequest.newBuilder().setDeleteAllDataFromTable(true));
```

Comment:
Is there a good reason to have a separate `issueBulkDeleteAsync` method instead of inline it here?